### DISCUSSION TITLE
[Explore]ability to search by column in datatable

### DISCUSSION THREAD

**Discussion by junlincc** [apache/superset#18438] [Ideas]
I want to look for a specific column and its sample data in a large dataset with 300+ columns. I can only search by the row values but not by column name, which is a major limitation of the search functionality...

<img width="1764" alt="Screen Shot 2021-02-06 at 6 01 47 PM" src="https://user-images.githubusercontent.com/67837651/107134371-5d0e2500-68a6-11eb-9868-1cca4ec64506.png">

**Comment by kgabryje** [comment]
What behaviour do we expect here? When a user types a search query, should the columns that don't match disappear? How do we want to tell if user wants to filter rows or columns?

**Comment by junlincc** [comment]
Expected behavior - only showing the matched columns in a table. 

but this is a low-priority. 😃 [[Explore]user can not type filter values when above 10k option limit](https://github.com/apache/superset/issues/13017)  is higher @kgabryje
